src/MainWindow.vala

```diff
@@ -998,18 +998,18 @@ namespace Terminal {
         }
 
         private void on_get_text (Gtk.Clipboard board, string? intext) {
-            /* if unsafe paste alert is enabled, show dialog */
-            if (Application.settings.get_boolean ("unsafe-paste-alert") && !unsafe_ignored ) {
+            if (intext == null) {
+                return;
+            }
 
-                if (intext == null) {
-                    return;
-                }
-                if (!intext.validate ()) {
-                    warning ("Dropping invalid UTF-8 paste");
-                    return;
-                }
-                var text = intext.strip ();
+            if (!intext.validate ()) {
+                debug ("Dropping invalid UTF-8 paste");
+                return;
+            }
+
+            unowned string text = intext._strip ();
```

Comment:
Why not just remove any trailing new line?

Reply:
Leading newlines could also unintentionally execute commands, but if it feels safer to only trim newlines and only trailing ones, then this could be adapted to do that as well.

Reply:
We really need to know if we are pasting into the commandline or not.  If we are pasting into e.g. vim or an interactive interpreter (?) then we do not want to strip.

Reply:
@jeremypw that makes sense to me, but I'm unsure if/how we can do that.

Reply:
@cassidyjames  I have pushed a commit that only processes the pasted text if there is no foreground process. This fixes pasting into terminal editors like nano or vim.  Also, I noticed that just stripping the string is insufficient to prevent command execution when pasting into the shell as it does not remove embedded newlines.  If you really want to prevent this, you have to make sure you only paste one line (without the newline).  My commit also does this. The UnsafePasteDialog (and setting) then becomes redundant as it is only triggered if there is a newline.

Reply:
@jeremypw yeah I was wondering about embedded newlines as well… but I wonder if this would be too aggressive/different from what people expect. I wonder if we could combine the checking with `sudo` and checking for newlines under the unsafe paste protection? I'll have to think about this a bit. But my initial thought is maybe:

- Treat newlines similar to how we treat `sudo` for paste protection, i.e.
  - Warn with an unsafe paste dialog (with modified copy) if a pasted command contains a newline
  - Turning unsafe paste protection off also turns off any special handling with newlines
- Maybe offer a way to paste one line without newlines in the dialog, i.e. as an alternate action? If that seems useful, we could also maybe strip `sudo` out from the sudo version, but idk if that's getting too complicated now.

Reply:
Maybe better to take an incremental approach and deal with single line pastes first (probably 99% of cases?) and think about how to deal with embedded newlines at leisure.